Assets/Scripts/Enemies/BossBehaviorController.cs

```diff
@@ -4,108 +4,156 @@
-        if(actionQueue.Any()) { currentAction =  actionQueue[0]; actionQueue.RemoveAt(0); }
-        else if(bossPhases[currentPhase].PossibleActions.Count > 1) 
-            currentAction = bossPhases[currentPhase].PossibleActions.Where(x=> !x.Equals(currentAction)).ElementAt( rand.Next(bossPhases[currentPhase].PossibleActions.Count-1) );
+    public void ChangeAction()
+    {
+        foreach(PhaseAction p in actionQueue) print(p.behavior);
+        if (currentAction.behavior != null) currentAction.behavior.enabled = false;
+        if (bossPhases[currentPhase].repeatingBehaviour != null && currentAction.behavior != bossPhases[currentPhase].repeatingBehaviour) currentAction.behavior = bossPhases[currentPhase].repeatingBehaviour;
+        else if (actionQueue.Any()) { currentAction = actionQueue[0]; actionQueue.RemoveAt(0); }
+        else if (bossPhases[currentPhase].PossibleActions.Count > 1)
+            currentAction = bossPhases[currentPhase].PossibleActions.Where(x => !x.Equals(currentAction)).ElementAt(rand.Next(bossPhases[currentPhase].PossibleActions.Count));
         else currentAction = bossPhases[currentPhase].PossibleActions.FirstOrDefault();
-
-        if(currentAction.behavior != null) StartCoroutine(startAction(currentAction, true));
+        if (currentAction.behavior != null) StartCoroutine(startAction(currentAction, true));
         else Debug.Log("BBC Error: Cannot start null behavior!");
     }
-    public IEnumerator startAction (PhaseAction action, bool startNewOnFinish) {
+    public IEnumerator startAction(PhaseAction action, bool startNewOnFinish)
+    {
         action.behavior.enabled = true;
+        action.behavior.actionRunning = true;
         int startedPhase = currentPhase;
-        for(int i = 0; i < action.runFrames; i++) {
-            if(currentPhase != startedPhase) {
-                action.behavior.enabled = false;
-                yield break;
+        if (action.independent || action.behavior == bossPhases[currentPhase].repeatingBehaviour)
+        {
+            while (action.behavior.actionRunning)
+            {
+                yield return new WaitForEndOfFrame();
             }
-            yield return new WaitForEndOfFrame();
-        } 
-        action.behavior.enabled = false;
-        for(int i = 0; i < action.cooldownFrames; i++) {
-            if(currentPhase != startedPhase) yield break;
+            action.behavior.enabled = false;
+
+        }
+        else
+        {
+            for (int i = 0; i < action.runFrames; i++)
+            {
+                if (currentPhase != startedPhase)
+                {
+                    action.behavior.enabled = false;
+                    yield break;
+                }
+                yield return new WaitForEndOfFrame();
+            }
+            action.behavior.enabled = false;
+        }
+        for (int i = 0; i < ((action.behavior == bossPhases[currentPhase].repeatingBehaviour) ? 0 : action.cooldownFrames); i++)
```

Comment:
why is this necessary

Reply:
looking back i dont think it is, i think originally i had it out there but then decided to move it inside the if statement, still it be easiest to just put the action.behavior.enabled = false; after the if else statement